WordPress/src/main/java/org/wordpress/android/ui/sitecreation/usecases/FetchHomePageLayoutsUseCase.kt

```diff
@@ -0,0 +1,42 @@
+package org.wordpress.android.ui.sitecreation.usecases
+
+import org.greenrobot.eventbus.Subscribe
+import org.greenrobot.eventbus.ThreadMode
+import org.wordpress.android.fluxc.Dispatcher
+import org.wordpress.android.fluxc.generated.ThemeActionBuilder
+import org.wordpress.android.fluxc.store.ThemeStore
+import org.wordpress.android.fluxc.store.ThemeStore.FetchStarterDesignsPayload
+import org.wordpress.android.fluxc.store.ThemeStore.OnStarterDesignsFetched
+import org.wordpress.android.ui.sitecreation.theme.ThumbDimensionProvider
+import javax.inject.Inject
+import kotlin.coroutines.Continuation
+import kotlin.coroutines.resume
+import kotlin.coroutines.suspendCoroutine
+
+class FetchHomePageLayoutsUseCase @Inject constructor(
+    val dispatcher: Dispatcher,
+    @Suppress("unused") val themeStore: ThemeStore,
+    private val thumbDimensionProvider: ThumbDimensionProvider
+) {
+    private var continuation: Continuation<OnStarterDesignsFetched>? = null
+
+    suspend fun fetchStarterDesigns(): OnStarterDesignsFetched {
+        if (continuation != null) {
+            throw IllegalStateException("Fetch already in progress.")
+        }
+        val payload = FetchStarterDesignsPayload(
+                thumbDimensionProvider.width.toFloat(),
```

Comment:
I noticed we're differing a bit on how we're sending the width to the REST call. I see here we're using the [phone's width](https://github.com/wordpress-mobile/WordPress-Android/blob/7712041199a52d979803f77018d90f9c6475b3da/WordPress/src/main/java/org/wordpress/android/ui/sitecreation/theme/ThumbDimensionProvider.kt#L38-L39), on my Pixel 3a it's asking for preview images with a width of 440 in portrait and 474 in landscape. 

On iOS, I'm using the size of the cell that display's the thumbnail, which makes a preview width of 320 on all devices with a scale of @2x and 480 with a scale of @3x. I think using the thumbnail's size might provide a better experience because then we can cache the data from portrait to landscape. And on larger devices like a tablet it might help keep the thumbnail sizes smaller. WDYT?

Reply:
I feel my solution is over-engineered but I couldn't find a simpler way to encode all the [design constrains](https://mobilegutenpagesp2.wordpress.com/2020/10/14/home-page-layout-picker-i2/).
* On **tablet and phone landscape** I'm using an `160dp` fixed width and translate that to pixels for the request.
* On **phone-portrait mode** thumbnail size dynamically to fill the full device width with two thumbnails per row.
* I always use `scale = 1.0` since I use pixels and not logical units in the call

> On iOS, I'm using the size of the cell that display's the thumbnail, which makes a preview width of 320 on all devices with a scale of @2x and 480 with a scale of @3x

I'm not sure if I can do the same on Android since the screen sizes are not standard.
Keeping a fixed preview width (especially in *phone-portrait mode*) will make things a lot simpler but will probably degrade the image quality in high resolution devices since the two thumbnails need to fill the full device width. Wdyt?

> we can cache the data from portrait to landscape

With the [addition of handling saved state](fd94213dfe44965e5f910f2ca57cfaa9aa1283ff) I prevent the data from reloading and use the already fetched images. This is not completely correct since it will use portrait size images for landscape and vice versa but seems better for user experience.

Reply:
> On tablet and phone landscape I'm using an 160dp fixed width and translate that to pixels for the request.
On phone-portrait mode thumbnail size dynamically to fill the full device width with two thumbnails per row.

Thanks for adding this context. I think I missed that part originally when looking at the code 👍